libsupport/include/katana/ArrowInterchange.h

```diff
@@ -279,12 +279,22 @@ TableBuilder::AddColumn(const ColumnOptions& options) {
 ////////////////////////////////////////////
 // Arrow utilities
 
-KATANA_EXPORT std::shared_ptr<arrow::Array> Unchunk(
+KATANA_EXPORT Result<std::shared_ptr<arrow::Array>> Unchunk(
     const std::shared_ptr<arrow::ChunkedArray>& original);
-KATANA_EXPORT std::shared_ptr<arrow::ChunkedArray> Shuffle(
+KATANA_EXPORT Result<std::shared_ptr<arrow::Array>> ScalarVecToArray(
+    const std::shared_ptr<arrow::DataType>& data_type,
+    const std::vector<std::shared_ptr<arrow::Scalar>>& data);
+KATANA_EXPORT Result<std::shared_ptr<arrow::ChunkedArray>> Shuffle(
     const std::shared_ptr<arrow::ChunkedArray>& original);
-KATANA_EXPORT std::shared_ptr<arrow::ChunkedArray> EmptyChunkedArray(
+KATANA_EXPORT std::shared_ptr<arrow::ChunkedArray> NullChunkedArray(
     const std::shared_ptr<arrow::DataType>& type, int64_t length);
+KATANA_EXPORT void PrintFirstNonEqualElements(
```

Comment:
BTW:
https://github.com/apache/arrow/blob/master/cpp/src/arrow/compare.h

If we really wanted to limit the output, we could just use a bounded output stream.

Reply:
The code you reference is just about equality testing.  My code prints some of the different elements and their indices, which I find helpful.  The report is pretty bounded (e.g., 1 or 2 lines per property), so I'm not worried about excessive length.  Do you want a bounded output stream?